lib/rmt/cli/base.rb

```diff
@@ -67,6 +67,14 @@ def handle_exceptions
         "The SCC credentials are not configured correctly in '/etc/rmt.conf'. You can obtain them from https://scc.suse.com/organization",
         RMT::CLI::Error::ERROR_SCC
       )
+    rescue RMT::Lockfile::ExecutionLockedError
+      raise RMT::CLI::Error.new(
+        "Process is locked. Please check lockfile at #{RMT::Lockfile::LOCKFILE_LOCATION}\n",
+        RMT::CLI::Error::ERROR_OTHER
+      )
+    rescue Exception # rubocop:disable RSpec/NestedGroups
+      RMT::Lockfile.remove_file
```

Comment:
Is this path covered by the specs?

Reply:
Why are you catching all exceptions here? This doesn't seem right.

Reply:
@wstephenson this is covered here https://github.com/SUSE/rmt/blob/78ceb1db9ea029247fc4a048a5d76de8bd0375cc/spec/lib/rmt/cli/main_spec.rb#L92

@ikapelyukhin if we don't catch all exceptions and remove lockfile it will stay if we have any exception that we don't catch. also I re-raise the original exception again after removing lockfile

Reply:
@Sergeykot see the previous PR and use the `ensure` block for removing the lock files.

Reply:
@ikapelyukhin in the 'ensure' block we unfortunately can not check for the exception class so that the lockfile would be removed even if we hit `RMT::Lockfile::ExecutionBlockedError`

Reply:
In the `ensure` block you don't need to check anything.

This `remove_file` call needs to be moved to `ensure` block around `locked_method` method  -- this will ensure that the lockfile is always removed, even when there's an exception raised:

https://github.com/SUSE/rmt/blob/0a7cd6660bccc728983c0b2cedce4c6343d26f2e/lib/rmt/cli/base.rb#L88-L91

Any other `remove_file` calls should be removed, like this one:

https://github.com/SUSE/rmt/blob/0a7cd6660bccc728983c0b2cedce4c6343d26f2e/lib/rmt/cli/base.rb#L94-L97

Or this one:

https://github.com/SUSE/rmt/blob/78ceb1db9ea029247fc4a048a5d76de8bd0375cc/lib/rmt/cli/base.rb#L75-L77